predict.py

```diff
@@ -10,26 +11,52 @@
     StableDiffusionPipeline,
 )
 
+# Stable Diffusion Defaults
 MODEL_ID = "stabilityai/stable-diffusion-2-1"
 MODEL_CACHE = "diffusers-cache"
+DEFAULT_HEIGHT = 768
+DEFAULT_WIDTH = 768
+DEFAULT_SCHEDULER = "K_EULER"
+DEFAULT_PROMPT = "a photo of an astronaut riding a horse on mars"
+
+# Dreambooth Training Defaults
+USE_DREAMBOOTH_WEIGHTS = os.path.exists("weights")
+if USE_DREAMBOOTH_WEIGHTS:
+    DEFAULT_HEIGHT = 512
+    DEFAULT_WIDTH = 512
+    DEFAULT_SCHEDULER = "DDIM"
+    try:
+        with open("weights/args.json") as f:
+            args = json.load(f)
+            if args.get('instance_prompt'):
+                DEFAULT_PROMPT = args['instance_prompt']
+    except:
+        pass
 
 
 class Predictor(BasePredictor):
     def setup(self):
         """Load the model into memory to make running multiple predictions efficient"""
         print("Loading pipeline...")
-        self.pipe = StableDiffusionPipeline.from_pretrained(
-            MODEL_ID,
-            cache_dir=MODEL_CACHE,
-            local_files_only=True,
-        ).to("cuda")
+        if USE_DREAMBOOTH_WEIGHTS:
+            self.pipe = StableDiffusionPipeline.from_pretrained(
+                "weights",
+                safety_checker=None,
+                torch_dtype=torch.float16,
```

Comment:
Might we worth adding some comments here to explain why the safety_checker is disabled, and why we're using float16?

Reply:
From https://github.com/replicate/cog-stable-diffusion/issues/14#issuecomment-1347671040

> Stable Diffusion 2.0 no longer has the NSFW filter because it was trained on data without NSFW content in it. It's here: https://replicate.com/stability-ai/stable-diffusion

Are we assuming here that most future DreamBooth weights are going to be using SD 2.x? Or is there some other reason for disabling the safety checker?

Reply:
k - I was just copying https://github.com/replicate/dreambooth-template/blob/main/predict.py#L23-L28

Definitely should comment - not sure why safety checker is turned off for dreambooth-template though

Reply:
cc @bfirsh @chenxwh